File: generated_code/optimized/memory/java/9.java
import java.util.*;
import java.lang.*;

class Solution {
    public List<Integer> rollingMax(List<Integer> numbers) {
        int size = numbers.size();
        List<Integer> result = new ArrayList<>(size);
        if (size == 0) {
            return result;
        }

        Iterator<Integer> it = numbers.iterator();
        int rollingMax = it.next();
        result.add(rollingMax);

        while (it.hasNext()) {
            int value = it.next();
            if (value > rollingMax) {
                rollingMax = value;
            }
            result.add(rollingMax);
        }

        return result;
    }
}

File: generated_code/optimized/runtime/detailed/cpp/9.cpp
#include <vector>
using namespace std;

vector<int> rolling_max(vector<int> numbers) {
    const size_t n = numbers.size();
    for (size_t i = 1; i < n; ++i) {
        if (numbers[i] < numbers[i - 1]) {
            numbers[i] = numbers[i - 1];
        }
    }
    return numbers;
}
